### PR TITLE
Handle undefined `message`

### PR DESCRIPTION
**Pull request by mariotsi**
From our remote logs we saw errors coming from this line `TypeError: Cannot read property 'startsWith' of undefined`. We are unable to check more deeply what was the response because we do not not log payloads. I'm making this change in order to make the code more failsafe

_Description of changes:_

Default the error message to empty string in order to make `.startsWith` not throwing in any case

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by mariotsi** [comment]
I see a Unit Test failing but I cannot see the link with my PR. Any help appreciated

**Comment by ericclemmons** [comment]
@mariotsi Thanks for your contribution!  I'm going to do some validation around why `error.message` is `undefined`.

In either case, you're right that we shouldn't be calling `.startsWith` non a non-String.

**Comment by ericclemmons** [comment]
Doing some research into this to determine why `err.message` would be empty to begin with: this `.startsWith` fix came from #3589 to address #1378.

We may need to update the test in https://github.com/aws-amplify/amplify-js/blob/master/packages/analytics/__tests__/Providers/AWSPinpointProvider-unit-test.ts#L737 to include an error without `message` set, to match your scenario.

File: packages/analytics/src/Providers/AWSPinpointProvider.ts [hunk 1]
```diff
@@ -447,7 +447,7 @@ export default class AWSPinpointProvider implements AnalyticsProvider {
 		failureData: EndpointFailureData
 	) {
 		const { err, update_params, endpointObject } = failureData;
-		const { message } = err;
+		const { message = '' } = err;
 		const { ApplicationId, EndpointRequest } = update_params;
 
 		if (!message.startsWith('Exceeded maximum endpoint per user count')) {
```
Comment (ericclemmons) [inline packages/analytics/src/Providers/AWSPinpointProvider.ts]: We're going to review this change and confirm the desired logic in the event an `err` comes back without a `message`.

One option is to not default `message` at all, and do a false-y check like so:

```suggestion
		if (!(message && message.startsWith('Exceeded maximum endpoint per user count'))) {
```

Going to mark this for review by the team & get this resolved.

Thanks for sharing your logs: I haven't seen an `err` without a `message` myself 🤞

Comment (ericclemmons) [reply packages/analytics/src/Providers/AWSPinpointProvider.ts]: I'm going to update this branch today with a test to validate the changes, then we should be good.

Thanks again for bringing this to our attention @mariotsi.  We've identified this is a `400`, but don't know the full shape of the response.

If you ever _do_ see the shape of the error (e.g. `{ Debug: "Whoopsie!" }`), we can add that into the test.

**Comment by ericclemmons** [comment]
@mariotsi I wasn't able to push directly to your branch, but I _did_ pull in your commits into a new PR that includes tests:

https://github.com/aws-amplify/amplify-js/pull/4507

I'd appreciate your 👀 on it: the only difference is that, since we can't guarantee the presence of `message` or that it's a string (vs. `null` or even a number), I had to be a bit more defensive.

**Pull request closed without merging**